pkg/services/alerting/service.go

```diff
@@ -130,6 +151,10 @@ func (s *AlertNotificationService) GetAlertNotificationsWithUidToSend(ctx contex
 func (s *AlertNotificationService) createNotifier(ctx context.Context, model *models.AlertNotification, secureSettings map[string]string) (Notifier, error) {
 	secureSettingsMap := map[string]string{}
 
+	if util.IsShortUIDTooLong(model.Uid) {
+		return nil, ValidationError{Reason: "Invalid UID: Must 40 characters or less"}
+	}
+
```

Comment:
Random nit for the code below. A bit confusing that those who calls `validateAlertNotification` which calls `createNotifier` already have encrypted the secrets, but I guess it's needed for the case when you load an existing alert notification and need to overwrite secure settings with updated ones.